spec/replication/replication.spec.ts

```diff
@@ -0,0 +1,86 @@
+import request from 'supertest';
+
+import { BaseEntity } from '../base/base.entity';
+import { BaseModule } from '../base/base.module';
+import { TestHelper } from '../test.helper';
+
+describe('replication', () => {
+    let app: INestApplication;
+    beforeAll(async () => {
+        const [moduleFixture] = await Promise.all([
+            Test.createTestingModule({
+                imports: [
+                    BaseModule,
+                    TypeOrmModule.forRoot({
+                        type: 'postgres',
+                        entities: [BaseEntity],
+                        synchronize: true,
+                        logging: true,
+                        logger: 'file',
+                        replication: {
+                            master: {
+                                database: process.env.POSTGRESQL_DATABASE_NAME,
+                                username: process.env.POSTGRESQL_DATABASE_USERNAME,
+                                password: process.env.POSTGRESQL_DATABASE_PASSWORD,
+                            },
+                            slaves: [
+                                {
+                                    port: Number(process.env.POSTGRESQL_SLAVE_PORT),
+                                    database: process.env.POSTGRESQL_DATABASE_NAME,
+                                    username: process.env.POSTGRESQL_DATABASE_USERNAME,
+                                    password: process.env.POSTGRESQL_DATABASE_PASSWORD,
+                                },
+                            ],
+                        },
+                    }),
+                ],
+            }).compile(),
+            await Test.createTestingModule({
+                imports: [
+                    BaseModule,
+                    TypeOrmModule.forRoot({
+                        type: 'postgres',
+                        entities: [BaseEntity],
+                        port: Number(process.env.POSTGRESQL_SLAVE_PORT),
+                        database: process.env.POSTGRESQL_DATABASE_NAME,
+                        username: process.env.POSTGRESQL_DATABASE_USERNAME,
+                        password: process.env.POSTGRESQL_DATABASE_PASSWORD,
+                        synchronize: true,
+                        logging: true,
+                        logger: 'file',
+                    }),
+                ],
+            }).compile(),
+        ]);
+        app = moduleFixture.createNestApplication();
+        await app.init();
+    });
+
+    afterAll(async () => {
+        await TestHelper.dropTypeOrmEntityTables();
+        await app?.close();
+    });
+
+    it('should not be affected by replication-lag', async () => {
+        const name = 'replication-test';
+        const { body: created } = await request(app.getHttpServer()).post('/base').send({ name }).expect(HttpStatus.CREATED);
+
+        await request(app.getHttpServer()).get(`/base/${created.id}`).expect(HttpStatus.NOT_FOUND);
```

Comment:
I want to check one thing quickly. Shouldn't we expect `OK` response when we request an Entity that we have just created?
If there is any misunderstanding, please let me know.

Reply:
This is because `replication-lag` is assumed in the environment.

Generally, it will operate in the following flow.
```
create --> master node -- replication --> slave node
findOne --------------------------------> slave node
```

If replication-lag occurs, 
created entity exists in the master node does not exist in the slave.

Reply:
when used replication in typeorm, 
search query(find, findOne...) send to slave node

Reply:
Now it's clear 👍🏼.
For readability, how about separating test blocks?
(I'm pretty sure there might be another person like me 🤣)

For example,
```ts
it('is not able to read lately created entity if replacation lag exists', () => {
    const name = 'replication-test';
    const { body: created } = await request(app.getHttpServer()).post('/base').send({ name }).expect(HttpStatus.CREATED);

    await request(app.getHttpServer()).get(`/base/${created.id}`).expect(HttpStatus.NOT_FOUND);
})

it('should be able to get result of lately updated or deleted entity even if replacation lag exists', () => {
    const name = 'replication-test';
    const { body: created } = await request(app.getHttpServer()).post('/base').send({ name }).expect(HttpStatus.CREATED);
    
    const { body: updated } = await request(app.getHttpServer())
        .patch(`/base/${created.id}`)
        .send({ name: 'new Name' })
        .expect(HttpStatus.OK);
        expect(updated.id).toEqual(created.id);
        expect(updated.name).toEqual('new Name');

    const { body: deleted } = await request(app.getHttpServer()).delete(`/base/${created.id}`).expect(HttpStatus.OK);
    expect(deleted.id).toEqual(created.id);

    const { body: recovered } = await request(app.getHttpServer()).post(`/base/${created.id}/recover`).expect(HttpStatus.CREATED);
    expect(recovered.id).toEqual(created.id);
})
```

Reply:
ok, i`ll reflect it :)

Reply:
Maybe we can use `beforeEach()` to reduce repeated codes here.

```ts
beforeEach(() => {
    const name = 'replication-test';
    const { body: created } = await request(app.getHttpServer()).post('/base').send({ name }).expect(HttpStatus.CREATED);
})
```

Reply:
oh,, I saw last comment late. Next time, i`ll take the time to refactor the spec. :)